src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Javier DELGADO
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.utility.steps.conf;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.DumperOptions;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.Yaml;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.io.*;
+import java.net.URL;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.*;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+/**
+ * Writes a yaml file from the workspace.
+ *
+ * @author Javier DELGADO &lt;witokondoria@gmail.com&gt;.
+ */
+public class WriteYamlStep extends AbstractStepImpl {
+
+    private String file;
+    private Object data;
+
+    @DataBoundConstructor
+    public WriteYamlStep(@Nonnull String file, @Nonnull Object data) {
+        if (file == null) {
+            throw new IllegalArgumentException("file parameter must be provided to writeYaml");
+        }
+        this.file = file;
```

Comment:
want to trim the string? Forcing a path from within the current current workspace seems fine